src/PIL/IcnsImagePlugin.py

```diff
@@ -310,6 +310,8 @@ def _save(im, fp, filename):
 
     # create the temporary set of pngs
     iconset = tempfile.mkdtemp('.iconset')
+    providedImages = {im.width:im for im in
```

Comment:
Shall we follow conventions and name it something like `provided_images`?

Reply:
Okay, sure. I've made that change.